examples/kubectl.rs

```diff
@@ -0,0 +1,169 @@
+//! This is a simple imitation of the basic functionality of kubectl
+//! Supports kubectl {get, delete, watch} <resource> [name] (name optional) with labels and namespace selectors
+use anyhow::{bail, Result};
+use clap::{arg, command};
+use either::Either;
+use futures::{StreamExt, TryStreamExt};
+use k8s_openapi::chrono::{Duration, Utc};
+use kube::{
+    api::{Api, DynamicObject, ListParams, ObjectMeta, Resource, ResourceExt},
+    discovery::{ApiCapabilities, ApiResource, Discovery, Scope},
+    runtime::{
+        utils::try_flatten_applied,
+        wait::{await_condition, conditions::is_deleted},
+        watcher,
+    },
+    Client,
+};
+use log::info;
+use std::{collections::BTreeMap, env};
+
+type KindMap = BTreeMap<String, (ApiResource, ApiCapabilities)>;
+async fn discover_valid_inputs(client: Client) -> Result<KindMap> {
+    let mut kinds = BTreeMap::new(); // valid inputs (keys are plural or kind)
+    let discovery = Discovery::new(client).run().await?;
+    // NB: relies on sort order of discovery groups for precedence (which is not perfect)
+    for group in discovery.groups() {
+        for (ar, caps) in group.recommended_resources() {
+            // two entries per (TODO: extend ApiResource with shortname - its in APIResource..)
+            kinds
+                .entry(ar.kind.to_lowercase())
+                .or_insert((ar.clone(), caps.clone()));
+            kinds.entry(ar.plural.to_lowercase()).or_insert((ar, caps));
+        }
+    }
+    Ok(kinds)
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // 0. init
+    std::env::set_var("RUST_LOG", "info,kube=info");
+    env_logger::init();
```

Comment:
Any reason not to use tracing here for consistency with kube-rs itself?

Reply:
This is currently the standard we use in all examples to make it as simple as possible to understand. We could probably migrate it to tracing_subscriber::fmt if there's a simple format that doesn't take too long to init.

Reply:
Yeah, the equivalent for tracing would be `tracing_subscriber::fmt::init()`.

Reply:
Oh, i thought that would have all the json style metadata. That's nice. +1 on swappping out env_logger::init with that!

Reply:
Replaced all in https://github.com/kube-rs/kube-rs/pull/885/commits/db29c57cb34e3a553049e61a21cfa25633a353ca